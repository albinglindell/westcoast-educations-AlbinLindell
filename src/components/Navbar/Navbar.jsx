import React, {Fragment, useState} from 'react'
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';

import Modal from '../Modal/Modal'





function Navbar() {
  const [modalValue, setModalValue] = useState(false)
  const openAddModalWindow = ()=>{
    setModalValue(true)
  }

  
  return (
    <div>
      <header className='Header'>
        <nav className='HeaderNav'>
        <Link to={"/"}><h2 >Westcourst education</h2></Link>
            <ul>
               <li onClick={openAddModalWindow}>Lägg till lärare och kurser</li>
            </ul>
        </nav>
      </header>
      {modalValue && <Fragment>
        {ReactDOM.createPortal(
        <Modal closeModal={setModalValue}/>,
        document.querySelector('#modal-root'),
      )}
      </Fragment>}
    </div>
  )
}

export default Navbar
