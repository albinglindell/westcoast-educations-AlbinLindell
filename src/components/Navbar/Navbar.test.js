import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from './Navbar';


describe("Navbar Component", ()=>{
    test("successfull render",  () => {
        const utils = render(<Navbar />, {wrapper: MemoryRouter});
        expect(utils).toMatchSnapshot();
        expect(utils).toBeTruthy();
      });


    test("test if modal works", ()=>{
        render(<Navbar />, {wrapper: MemoryRouter});

        const addModal = screen.getByText("Lägg till lärare och kurser")

        expect(addModal).toBeInTheDocument

        //--------------------Vi fick aldrig lära oss om hur an testar portals, så här är ett tappert försök.

        // fireEvent.click(addModal)

        // const addBtn = screen.getByText("Lägg till utbildning")

        // expect(addBtn).toBeInTheDocument

    })
})