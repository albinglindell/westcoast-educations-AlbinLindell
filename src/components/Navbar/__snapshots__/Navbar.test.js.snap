// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component successfull render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="Header"
        >
          <nav
            class="HeaderNav"
          >
            <a
              href="/"
            >
              <h2>
                Westcourst education
              </h2>
            </a>
            <ul>
              <li>
                L채gg till l채rare och kurser
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="Header"
      >
        <nav
          class="HeaderNav"
        >
          <a
            href="/"
          >
            <h2>
              Westcourst education
            </h2>
          </a>
          <ul>
            <li>
              L채gg till l채rare och kurser
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
